query_optimizer/PhysicalGenerator.cpp

```diff
@@ -176,6 +177,8 @@ P::PhysicalPtr PhysicalGenerator::optimizePlan() {
     rules.emplace_back(new AttachLIPFilters());
   }
 
+  rules.push_back(std::make_unique<Partition>(optimizer_context_));
```

Comment:
Is it okay to move this rule before the two LIP-related rules? Currently the LIP information may be invalidated (and lead to incorrect results) if tree transformation happens after the two LIP passes.

Meanwhile, better add some comments describing what additional information would be added into the physical nodes after this pass.